### DISCUSSION TITLE
debug lua script

### DISCUSSION THREAD

**Discussion by shaqtsui** [xmake-io/xmake#3398] [Ideas]
I am wondering whether we can introduce a mechanism in  xamke to support lua debug?

I can see benifits here:
  1, As this tool is used by developers mostly, so we as developer can debug into the internal implementation to find the root cause, if we got some errors during the usage. Which means less trouble ruki :)
   2, Some bug fix may be easily tested before merge request.
   3, For those who would like to learn inside xmake, debugger can help a lot.

**Comment by shaqtsui** [comment]
If the direction is conciderable. The implementation should not a big task. Some remote debugger only need target to have luasocket included.

**Reply by waruqi** [reply]
I don't have time to work on it at the moment, because I just need print to finish debugging. If you're interested, you can try to make it support lua debugging, as far as I know lua 5.4 should support it.

**Reply by shaqtsui** [reply]
I have tried `require "socket.core"` or `import "socket.core"` in `xmake l` env. Confirmed that currently `socket.core` is not integrated. 
As I know luasocket is 3rd part package, by default not included in standard lua. 
I can work on this, but need your hand on add luasocket into lua runtime (luasocket is about 200KB) 
or make xmake able to use external lua interpreter(this approach is better as it's much more flexable).

**Reply by waruqi** [reply]
xmake has builtin socket module, `import("core.base.socket")`, we can not add luasocket.

**Reply by shaqtsui** [reply]
Is it possible to make xmake extensiable? Which means support `require` system lua packages?
If so, I can go on to integrate debugger.
